metadata-ingestion/build.gradle

```diff
@@ -66,15 +72,23 @@ task lint(type: Exec, dependsOn: installDev) {
    */
   commandLine 'bash', '-x', '-c',
     "find ${venv_name}/lib -path *airflow/_vendor/connexion/spec.py -exec sed -i.bak -e '169,169s/  # type: List\\[str\\]//g' {} \\; && " +
-    "source ${venv_name}/bin/activate && black --check --diff src/ tests/ examples/ && isort --check --diff src/ tests/ examples/ && flake8 --count --statistics src/ tests/ examples/ && mypy src/ tests/ examples/"
+    "source ${venv_name}/bin/activate && black --check --diff src/ tests/ examples/ --exclude src/datahub/ingestion/source/dist/ && isort --check --diff src/ tests/ examples/ -s src/datahub/ingestion/source/dist/ && flake8 --count --statistics src/ tests/ examples/ --exclude src/datahub/ingestion/source/dist/ && mypy src/ tests/ examples/ --exclude src/datahub/ingestion/source/dist/"
 }
 task lintFix(type: Exec, dependsOn: installDev) {
   commandLine 'bash', '-x', '-c',
     "source ${venv_name}/bin/activate && " +
-    "black src/ tests/ examples/ && " +
-    "isort src/ tests/ examples/ && " +
-    "flake8 src/ tests/ examples/ && " +
-    "mypy src/ tests/ examples/"
+    "black src/ tests/ examples/ --exclude src/datahub/ingestion/source/dist/ && " +
+    "isort src/ tests/ examples/ -s src/datahub/ingestion/source/dist/ && " +
+    "flake8 src/ tests/ examples/ --exclude src/datahub/ingestion/source/dist/ && " +
+    "mypy src/ tests/ examples/ --exclude src/datahub/ingestion/source/dist/"
+}
+
+task generateThriftGrammar(type: JavaExec) {
+  classpath = buildscript.configurations.classpath
+  mainClass = 'org.antlr.v4.Tool'
+  ignoreExitValue true
+  workingDir="src/datahub/ingestion/source"
```

Comment:
can we use a temporary directory instead, that is deleted after ingestion is complete?

Reply:
Sorry, I can't understand this, can you give me an example?

Reply:
You could use File.createTempDir and also specify that it be deleted on exit. Here is an example of how to do it.
https://github.com/sberbankjavaschool/homework/blob/11db332ee36bfa1f686842318a8ee68a7dd18c04/build.gradle#L77

Reply:
If this task is building the lexer/parser etc thrift tools, we can even create the temporary directory under the project's buildDir.